custom_components/opnsense/pyopnsense/__init__.py

```diff
@@ -853,193 +855,195 @@ def send_wol(self, interface, mac):
+            "five_minute": float(load_list[1]),
+            "fifteen_minute": float(load_list[2]),
+        }
+        _LOGGER.debug(f"[get_telemetry_system] system: {system}")
+        return system
 
-foreach ($ovpn_servers as $server) {
-    $vpnid = $server["vpnid"];
-    $name = $server["name"];
-    $conn_count = count($server["conns"]);
-    $total_bytes_recv = 0;
-    $total_bytes_sent = 0;
-    foreach ($server["conns"] as $conn) {
-        $total_bytes_recv += $conn["bytes_recv"];
-        $total_bytes_sent += $conn["bytes_sent"];
-    }
-    
-    $toreturn["openvpn"]["servers"][$vpnid]["name"] = $name;
-    $toreturn["openvpn"]["servers"][$vpnid]["vpnid"] = $vpnid;
-    $toreturn["openvpn"]["servers"][$vpnid]["connected_client_count"] = $conn_count;
-    $toreturn["openvpn"]["servers"][$vpnid]["total_bytes_recv"] = $total_bytes_recv;
-    $toreturn["openvpn"]["servers"][$vpnid]["total_bytes_sent"] = $total_bytes_sent;
-}
+    @_log_errors
+    def _get_telemetry_cpu(self) -> dict:
+        cpu: dict[str, Any] = {}
+        cputype_info: dict[str, Any] = self._post("/api/diagnostics/cpu_usage/getCPUType")
+        _LOGGER.debug(f"[get_telemetry_cpu] cpu_info: {cputype_info}")
+        # Hacky and probably not universally applicable
+        cores_match = re.search(r'\((\d+) cores', cputype_info[0])
+        cpu["count"] = int(cores_match.group(1)) if cores_match else 0
+        # Missing frequency current and max
+        # cpu["frequency"] = {"current": 0, "max": 0}
+        _LOGGER.debug(f"[get_telemetry_cpu] cpu: {cpu}")
+        return cpu
 
-"""
-        data = self._exec_php(script)
+    @_log_errors
+    def _get_telemetry_filesystems(self) -> dict:
+        filesystems: dict[str, Any] = {}
+        filesystems_info: dict[str, Any] = self._post(
+            "/api/diagnostics/system/systemDisk"
+        )
+        _LOGGER.debug(f"[get_telemetry_filesystems] filesystems_info: {filesystems_info}")
+        filesystems = filesystems_info["devices"]
+        # To conform to the previous data being returned
+        for filesystem in filesystems:
+            filesystem["size"] = filesystem.pop("blocks", None)
+            filesystem["capacity"] = f"{filesystem.pop("used_pct")}%"
+        _LOGGER.debug(f"[get_telemetry_filesystems] filesystems: {filesystems}")
+        return filesystems
 
-        if isinstance(data["gateways"], list):
-            data["gateways"] = {}
+    @_log_errors
+    def _get_telemetry_openvpn(self) -> dict:
+        openvpn: dict[str, Any] = {}
+        openvpn["servers"] = {}
+        openvpn_info: dict[str, Any] = self._post("/api/openvpn/export/providers")
+        connection_info: dict[str, Any] = self._post("/api/openvpn/service/searchSessions")
+        _LOGGER.debug(f"[get_telemetry_openvpn] openvpn_info: {openvpn_info}")
+        _LOGGER.debug(f"[get_telemetry_openvpn] connection_info: {connection_info}")
+        for vpnid, vpn_info in openvpn_info.items():
```

Comment:
Added a check if `openvpn_info` list is empty. Happens when there's no OpenVPN interface (at least here), and it stops the integration from loading. I modified it like this and it works now.

```python
    @_log_errors
    def _get_telemetry_openvpn(self) -> dict:
        openvpn: dict[str, Any] = {}
        openvpn["servers"] = {}
        openvpn_info: dict[str, Any] = self._post("/api/openvpn/export/providers")
        connection_info: dict[str, Any] = self._post(
            "/api/openvpn/service/searchSessions"
        )
        _LOGGER.debug(f"[get_telemetry_openvpn] openvpn_info: {openvpn_info}")
        _LOGGER.debug(f"[get_telemetry_openvpn] connection_info: {connection_info}")
        if openvpn_info:
            for vpnid, vpn_info in openvpn_info.items():
                vpn: dict[str, Any] = {}
                vpn["vpnid"] = vpn_info["vpnid"]
                vpn["name"] = vpn_info["name"]
                total_bytes_recv = 0
                total_bytes_sent = 0
                for connect in connection_info["rows"]:
                    if connect["id"] == vpn["vpnid"]:
                        total_bytes_recv += (
                            int(connect["bytes_received"])
                            if connect["bytes_received"]
                            else 0
                        )
                        total_bytes_sent += (
                            int(connect["bytes_sent"]) if connect["bytes_sent"] else 0
                        )
                vpn["total_bytes_recv"] = total_bytes_recv
                vpn["total_bytes_sent"] = total_bytes_sent
                # Missing connected_client_count
                # vpn["connected_client_count"] =
                openvpn["servers"][vpnid] = vpn
            _LOGGER.debug(f"[get_telemetry_openvpn] openvpn: {openvpn}")
        else:
            _LOGGER.debug("[get_telemetry_openvpn] openvpn_info: no OpenVPN interfaces")
        return openvpn
```